augur/tasks/util/collection_util.py

```diff
@@ -132,9 +132,10 @@ def get_required_conditions_for_ml_repos(allow_collected_before = False, days_un
 
 
 class CollectionRequest:
```

Comment:
̶W̶e̶ ̶s̶h̶o̶u̶l̶d̶ ̶p̶r̶o̶b̶a̶b̶l̶y̶ ̶c̶h̶a̶n̶g̶e̶ ̶t̶h̶i̶s̶ ̶i̶n̶t̶o̶ ̶s̶o̶m̶e̶ ̶k̶i̶n̶d̶ ̶o̶f̶ ̶s̶e̶p̶a̶r̶a̶t̶e̶d̶ ̶c̶l̶a̶s̶s̶ ̶s̶t̶r̶u̶c̶t̶u̶r̶e̶ ̶w̶h̶e̶r̶e̶ ̶w̶e̶ ̶h̶a̶v̶e̶ ̶a̶ ̶`̶G̶i̶t̶H̶u̶b̶C̶o̶l̶l̶e̶c̶t̶i̶o̶n̶R̶e̶q̶u̶e̶s̶t̶`̶ ̶a̶n̶d̶ ̶a̶ ̶`̶G̶i̶t̶L̶a̶b̶C̶o̶l̶l̶e̶c̶t̶i̶o̶n̶R̶e̶q̶u̶e̶s̶t̶`̶ ̶l̶a̶t̶e̶r̶ ̶b̶u̶t̶ ̶I̶ ̶t̶h̶i̶n̶k̶ ̶i̶t̶'̶s̶ ̶f̶i̶n̶e̶ ̶f̶o̶r̶ ̶n̶o̶w̶.̶

Reply:
Although, that might complicate our fairness algorithm since we treat GitHub and GitLab equally and it makes sense to mix them together in a `CollectionRequest`. Instead what we should do is use some kind of mapping logic so that the `CollectionRequest` can detect the platform of the repo and map it similar to [this example](https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/too-many-branches.html).